### ISSUE TITLE
Do we have KORE support for pretty-printing?

### ISSUE THREAD

**Issue by grosu** [kframework/k-legacy#1940]
Do we?  If we do, then wouldn't it be nice to use it when defining the `Id2String` function?  That is, instead of using a hook, like we do now,

```
syntax String ::= "Id2String" "(" Id ")"    [function, hook(STRING.token2string), klabel(Id2String)]
```

to have something like this

```
syntax String ::= Id2String(Id)  [function]
rule Id2String(X:Id) => #prettyPrint(X)
```

**Comment by cos** [comment]
Yes, we could.
Should `#prettyPrint` be hooked to the meta or ground-level unparsing? I.e., should it print `#token("foo", Id)` or `foo`? From the example, it seems it should be the latter.

**Comment by grosu** [comment]
Yes, I think the latter.  And this should be the naive prettyPrinter, not the smart unparser.  We will probably have an unparser hook, too, but no need to worry about it right now.  Thanks.

**Comment by dwightguth** [comment]
I will say I have no objection to having a #prettyPrint function but I am strongly opposed to using it to implement more basic conversions between types. This is both a performance issue and an issue with compatibility because supporting the pretty-printer from within the OCAML backend would be highly nontrivial.
